lib/utils/repo-page-utils.ts

```diff
@@ -1,8 +1,45 @@
 import differenceInDays from "date-fns/differenceInDays";
 import { type StatsType } from "lib/hooks/api/useFetchMetricStats";
 
+export type HistogramData = {
+  bucket: string;
+  star_count?: number;
+  forks_count?: number;
+};
+
+export function getTicks({ histogram, range }: { histogram: HistogramData[]; range: number }): string[] {
+  // if 30 days, get every 3 days
+  // if 90 days, get every week
+  // if 7 days, get all days
+  const result: string[] = [];
+  switch (range) {
+    case 7:
+      return histogram.map((point) => point.bucket);
+    case 30: {
+      for (let i = 3; i < 30; i += 3) {
+        result.push(histogram[i].bucket);
+      }
+      return result;
+    }
+    case 90: {
+      for (let i = 7; i < 90; i += 7) {
+        result.push(histogram[i].bucket);
+      }
+      return result;
+    }
+    default:
+      return [];
```

Comment:
If the range is incorrect, should we not throw an error? Or is this to skip 180 and 360 day ranges?

Reply:
I'm keeping it to an empty array so that it doesn't block the graph rendering at all.